pkg/router/controller/host_admitter.go

```diff
@@ -0,0 +1,204 @@
+package controller
```

Comment:
Why is it desirable to add a new plugin vs updating UniqueHost to support wildcard routes?  I don't understand why it wouldn't be simpler and less error prone to reuse the existing logic with normalized host values.